server/athenian/api/precompute/discover_accounts.py

```diff
@@ -28,5 +37,30 @@ async def main(context: PrecomputeContext, args: argparse.Namespace) -> List[int
         if state.finished_date is None:
             log.warning("Skipped account %d because the progress is not 100%%", account)
             continue
-        result.append(account)
-    return result
+
+        discovered["precomputed" if precomputed else "fresh"].append(account)
+
+    if args.partition:
+        return discovered
+    else:
+        return sorted(chain(*discovered.values()))
+
+
+async def _get_accounts(sdb: Database) -> List[Tuple[int, bool]]:
+    """Return the existing account IDs, each with a precomputed flag."""
+    # already precomputed accounts have an ALL repository set that is precomputed
+    # outer join will return one row per account, since (owner_id, name) is unique in RepositorySet
+    join_cond = sa.and_(
+        Account.id == RepositorySet.owner_id,
+        RepositorySet.name == RepositorySet.ALL,
```

Comment:
We should move this inside `WHERE`. Postgres doesn't really like when you put constant conditions inside JOIN, it produces unexpected, bad plans. `WHERE` is always bullet-proof, on the other hand, and always leads to good plans.

Reply:
need condition in `ON` to have accounts with reposets

Reply:
But wait, we've got a LEFT JOIN specifically because we may have accounts without reposets.

Reply:
True